.github/workflows/linux-eic-shell.yml

```
@@ -332,17 +332,21 @@ jobs:
           *.dot
           *.svg
         if-no-files-found: error
-    - uses: dawidd6/action-download-artifact@v2
+    - name: Download previous artifact
+      id: download_previous_artifact
+      uses: dawidd6/action-download-artifact@v2
       with:
         branch: ${{ github.event.pull_request.base.ref || github.ref_name }}
         path: ref/
         name: rec_${{ matrix.particle }}_1GeV_20GeV_${{ matrix.detector_config }}.edm4eic.root
         if_no_artifact_found: warn
     - uses: actions/checkout@v3
+      if: steps.download_previous_artifact.outputs.found_artifact == true
```

The booleans are a lie.
```suggestion
      if: steps.download_previous_artifact.outputs.found_artifact == 'true'
```